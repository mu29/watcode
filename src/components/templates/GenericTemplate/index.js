import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { palette } from 'services/style'

const Wrapper = styled.div`
  display: flex;
  min-height: 100vh;
  background-color: ${palette('white.default')};
`

const Header = styled.header`
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 999;
`

const Content = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
`

const GenericTemplate = ({ header, children, ...props }) => (
  <Wrapper { ...props }>
    <Header>{ header }</Header>
    <Content>{ children }</Content>
  </Wrapper>
)

GenericTemplate.propTypes = {
  header: PropTypes.node.isRequired,
  children: PropTypes.node.isRequired,
}

export default GenericTemplate
