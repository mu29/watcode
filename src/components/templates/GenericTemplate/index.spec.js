import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import GenericTemplate from '.'

const wrap = (props = { header: 'header' }) => shallow(<GenericTemplate { ...props }>test</GenericTemplate>)

describe('<GenericTemplate />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = wrap()
    expect(wrapper).toMatchSnapshot()
  })
})
