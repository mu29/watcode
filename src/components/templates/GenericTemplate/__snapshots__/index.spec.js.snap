// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenericTemplate /> 정상적으로 렌더링된다 1`] = `
<Wrapper>
  <Header>
    header
  </Header>
  <Content>
    test
  </Content>
</Wrapper>
`;
